modules/tinymce/src/plugins/directionality/test/ts/browser/DirectionalitySanityTest.ts

```diff
@@ -17,31 +17,308 @@ describe('browser.tinymce.plugins.directionality.DirectionalitySanityTest', () =
     editor.setContent('a');
     TinySelections.setSelection(editor, [ 0, 0 ], 0, [ 0, 0 ], 1);
     TinyUiActions.clickOnToolbar(editor, 'button[title="Right to left"]');
+    TinyAssertions.assertContent(editor, '<p dir="rtl">a</p>');
+  });
+
+  it('TBA: Set and select content, click on the Left to right toolbar button and assert direction is left to right', () => {
+    const editor = hook.editor();
+    editor.setContent('<p dir="rtl">a</p>');
+    TinySelections.setSelection(editor, [ 0, 0 ], 0, [ 0, 0 ], 1);
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Left to right"]');
+    TinyAssertions.assertContent(editor, '<p>a</p>'); // as the default dir is ltr it just removes the dir attr
+  });
+
+  it('TINY-4589: should set two paragraphs to rtl and ltl', () => {
+    const editor = hook.editor();
+    editor.setContent('<p>foo</p><p>bar</p>');
+    TinySelections.setSelection(editor, [ 0 ], 0, [ 1 ], 1);
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Right to left"]');
+    TinyAssertions.assertContent(editor, '<p dir="rtl">foo</p>\n<p dir="rtl">bar</p>');
+
+    TinySelections.setSelection(editor, [ 0 ], 0, [ 1 ], 1);
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Left to right"]');
+    TinyAssertions.assertContent(editor, '<p>foo</p>\n<p>bar</p>');
+  });
+
+  it('TINY-4589: should set parent dir when element is a list item', () => {
+    const editor = hook.editor();
+    editor.setContent('<ul><li>foo</li><li>bar</li></ul>');
+    TinySelections.setSelection(editor, [ 0, 0 ], 0, [ 0, 0 ], 1);
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Right to left"]');
+    TinyAssertions.assertContent(editor, '<ul dir="rtl">\n<li>foo</li>\n<li>bar</li>\n</ul>');
+
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Left to right"]');
+    TinyAssertions.assertContent(editor, '<ul>\n<li>foo</li>\n<li>bar</li>\n</ul>');
+  });
+
+  it('TINY-4589: should remove dir from selected list item and children', () => {
+    const editor = hook.editor();
+    editor.setContent(`
+    <ul>
+      <li dir="ltr">
+        foo
+        <ul>
+          <li dir="ltr">a</li>
+          <li dir="rtl">b</li>
+          <li>c</li>
+        </ul>
+      </li>
+      <li dir="xyz">bar</li>
+    </ul>
+    `);
+    TinySelections.setSelection(editor, [ 0, 0 ], 0, [ 0, 0 ], 1);
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Right to left"]');
     TinyAssertions.assertContentStructure(editor, ApproxStructure.build((s, str) => {
```

Comment:
I'm curious - could this be a string assertion instead of an ApproxStructure, and do we care?

Reply:
I suspect this is because of the indentation. This can be solved by adding `indent: false` to the editor settings, so yes we should use that instead.

- [ ] Please convert all these to a string based assertion instead